metamath-rs/src/grammar.rs

```diff
@@ -1540,6 +1540,7 @@ impl Grammar {
         }
     }
 
+    #[allow(clippy::missing_const_for_fn)] // false positive
```

Comment:
not FP

Reply:
see https://github.com/metamath/metamath-knife/pull/185/commits/2566a4aacfd78bf0014ba5585bbb5c1e1a42de73 CI error

Reply:
This sounds like https://github.com/rust-lang/rust-clippy/issues/14020 ? It may already be fixed. Does it work on nightly clippy? Rather than calling it a false positive, you should put a link or short link e.g. `rust-clippy#14020` as a comment on all the allows instead. If this is not the right error report, then we should open one and then link to it.

Reply:
It seems to be a mix of
https://github.com/rust-lang/rust-clippy/issues/14020
https://github.com/rust-lang/rust-clippy/issues/14091
and a case similar to non-const deref but a non-const destructor instead (I linked to the pr in this case)

all fixed by https://github.com/rust-lang/rust-clippy/pull/14294 for nightly